mpi-proxy-split/mpi-wrappers/mpi_type_wrappers.cpp

```diff
@@ -122,6 +122,20 @@ USER_DEFINED_WRAPPER(int, Type_vector, (int) count, (int) blocklength,
   return retval;
 }
 
+USER_DEFINED_WRAPPER(int, Type_hvector, (int) count, (int) blocklength,
+                     (MPI_Aint) stride, (MPI_Datatype) oldtype,
+                    (MPI_Datatype*) newtype)
+{
+  int size;
+  int retval = MPI_Type_size(oldtype, &size);
+  if(retval != MPI_SUCCESS){
+    return retval;
+  }
+  JASSERT(stride % size == 0).Text("Current implementation only supports "
```

Comment:
Could you add a comment on why we only support even stride? What is the issue with odd strides?

Reply:
By even multiple I mean that the stride must be a multiple of the size of the type (stride % size == 0). This allows us to use the MPI_Type_vector to implement this method. A cleaner implementation would be to use MPI_Type_hvector to implement MPI_Type_vector, but this would require significantly more changes right now (that risks breaking MPI_Type_vector), and this current implementation should be good enough for most cases.

Reply:
Got it. Please add the same description as a comment and a FIXME in the code. It would be helpful for whoever improves the code.